tests/unit/tabular/test_base.py

```diff
@@ -343,3 +344,38 @@ def test_sample_batches_transform_conditions_correctly():
     model._sample_batch.assert_any_call(
         1, 100, 10, {'column1': 30}, {'transformed_column': 70}, 0.01
     )
+
+
+def test__make_conditions_df_without_num_rows():
+    """Test ``_make_conditions_df`` works correctly when ``num_rows`` is not passed.
+
+    The ``_make_conditions_df`` method is expected to:
+    - Return conditions as a ``DataFrame`` for every row in the data.
+
+    Input:
+    - Conditions
+
+    Output:
+    - Conditions as ``DataFrame``
+    """
+    # Setup
+    N_ROWS = 1000
+    model = GaussianCopula()
+    data = pd.DataFrame({
+        'column1': list(range(N_ROWS)),
+        'column2': [random.choice(['M', 'F']) for _ in range(N_ROWS)],
+    })
+    conditions = {
+        'column2': 'M'
+    }
+    expected_conditions = pd.DataFrame([conditions] * len(data))
+
+    model.fit(data)
```

Comment:
Instead of calling `model.fit`, let's directly set `model._num_rows = N_ROWS`. That way we are more explicit with the set up, and the test is isolated to testing `_make_conditions_df`.

Reply:
Yes, that is a good idea.
I will check if there is a test that checks that '_num_rows' is set when fitting. If not I will add a test for this, to ensure this behavior is kept.